components/suggest/src/db.rs

```diff
@@ -152,58 +147,57 @@ impl<'a> SuggestDao<'a> {
                 match provider {
                     SuggestionProvider::Amp => {
                         self.conn.query_row_and_then(
-                            "SELECT amp.advertiser, amp.block_id, amp.iab_category, amp.impression_url, amp.click_url
-                            FROM amp_custom_details amp WHERE amp.suggestion_id = :suggestion_id", 
+                            "SELECT amp.advertiser, amp.block_id, amp.iab_category, amp.impression_url, amp.click_url,
+                                    (SELECT i.data FROM icons i WHERE i.id = amp.icon_id) AS icon
+                             FROM amp_custom_details amp
+                             WHERE amp.suggestion_id = :suggestion_id",
                             named_params! {
                                 ":suggestion_id": suggestion_id
                             },
-                            |row|{
-                                let iab_category = row.get::<_, String>("iab_category")?;
-                                let is_sponsored = !NONSPONSORED_IAB_CATEGORIES.contains(&iab_category.as_str());
-                                Ok(Suggestion {
+                            |row| {
+                                Ok(Suggestion::Amp {
                                     block_id: row.get("block_id")?,
                                     advertiser: row.get("advertiser")?,
-                                    iab_category,
-                                    is_sponsored,
+                                    iab_category: row.get("iab_category")?,
                                     title,
                                     url,
                                     full_keyword: full_keyword(keyword, &keywords),
-                                    icon,
+                                    icon: row.get("icon")?,
                                     impression_url: row.get("impression_url")?,
                                     click_url: row.get("click_url")?,
-                                    provider
                                 })
                             }
                         )
                     },
-                    SuggestionProvider::Wikipedia =>{
-                        Ok(Suggestion {
-                            block_id: 0,
-                            advertiser: "Wikipedia".to_string(),
-                            iab_category: "5 - Education".to_string(),
-                            is_sponsored: false,
+                    SuggestionProvider::Wikipedia => {
+                        let icon = self.conn.try_query_one::<Vec<u8>, _>(
+                            "SELECT i.data AS icon
```

Comment:
Thanks for pointing that out! It was part of the reason I moved `icon_id` into the AMP and Wikipedia tables 😊 I noticed that AMO suggestions use `https://` URLs for icons, instead of including the icon data directly; and Pocket suggestions didn't have icons in Remote Settings at all.

Android and iOS can load icons from `https://` and `data:` URLs (yay!) but not `chrome://` URLs (boo!): on Android, the awesomebar code is "pure" Android, outside of GeckoView, and iOS doesn't have Gecko at all.

Reply:
Gotcha! Thanks for that note! We use those TippyTop icons (`chrome://`) for Dynamic Wikipedia and Top Picks for desktop, definitely need to revisit that once we bring those providers to x-platform.

Reply:
Filed https://mozilla-hub.atlassian.net/browse/DISCO-2573 for our future selves, thanks! 😊